src/assets/scripts/client/aircraft/AircraftController.js

```diff
@@ -568,6 +641,80 @@ export default class AircraftController {
     }
 
     /**
+     * Generate a unique `transponderCode`
+     *
+     * This method should only be run while building props for a
+     * soon-to-be-instantiated `AircraftModel`
+     *
+     * @for AircraftController
+     * @method _generateUniqueTransponderCode
+     * @return {number}
+     * @private
+     */
+    _generateUniqueTransponderCode() {
+        const minCodeValue = 1203;
+        const maxCodeValue = 7499;
+        const transponderCode = _random(minCodeValue, maxCodeValue);
+
+        if (!this._isValidTransponderCode(transponderCode)) {
+            // the value generated is already in use, recurse back through this method and try again
+            this._generateUniqueTransponderCode();
```

Comment:
what are your concerns with recursion?  using recursion is actually very efficient in javascript, especially in this case.

It also keeps close to a pattern used in a few other places, so if we wanted to abstract the logic here we could.

As far as performance, I would argue that using a recursion strategy is slightly faster.  But overall, I wouldn't be surprised if either strategy worked out to about the same performance.  I just feel that this reads better and is easier to maintain (plus the aforementioned pattern)